wrappers/azurekeyvault/azurekeyvault.go

```diff
@@ -287,31 +353,80 @@ func (v *Wrapper) buildBaseURL() string {
 	return fmt.Sprintf("https://%s.%s/", v.vaultName, v.environment.KeyVaultDNSSuffix)
 }
 
-func (v *Wrapper) getKeyVaultClient(withCertPool *x509.CertPool) (*azkeys.Client, error) {
-	var err error
-	var cred azcore.TokenCredential
+func (v *Wrapper) getCredential() (azcore.TokenCredential, error) {
```

Comment:
and this would take `authMethod` as an explicit parameter and not use `v.authMethod` internally.